catroid/src/main/java/org/catrobat/catroid/transfers/MediaDownloadService.java

```diff
@@ -24,85 +24,55 @@
 import android.os.ResultReceiver;
 import android.util.Log;
 
-import org.catrobat.catroid.R;
-import org.catrobat.catroid.common.Constants;
-import org.catrobat.catroid.utils.ToastUtil;
+import org.catrobat.catroid.utils.StatusBarNotificationManager;
+import org.catrobat.catroid.web.ProgressResponseBody;
 import org.catrobat.catroid.web.ServerCalls;
-import org.catrobat.catroid.web.WebconnectionException;
 
-import java.io.IOException;
+import static org.catrobat.catroid.common.Constants.UPDATE_DOWNLOAD_ERROR;
+import static org.catrobat.catroid.common.Constants.UPDATE_DOWNLOAD_SUCCESS;
 
 public class MediaDownloadService extends IntentService {
 
 	public static final String TAG = MediaDownloadService.class.getSimpleName();
 
-	public static final String RECEIVER_TAG = "receiver";
 	public static final String URL_TAG = "url";
 	public static final String MEDIA_FILE_PATH = "path";
-
-	public ResultReceiver receiver;
-	private Handler handler;
+	public static final String PROGRAM_NAME = "programname";
+	public static final String RECEIVER = "receiver";
 
 	public MediaDownloadService() {
 		super(MediaDownloadService.class.getSimpleName());
 	}
 
-	@Override
-	public void onCreate() {
-		super.onCreate();
-		handler = new Handler();
-	}
-
 	@Override
 	protected void onHandleIntent(Intent intent) {
-		boolean result = true;
 		String url = intent.getStringExtra(URL_TAG);
 		String fileString = intent.getStringExtra(MEDIA_FILE_PATH);
-		int errorMessage = R.string.error_unknown_error;
-
-		receiver = intent.getParcelableExtra(RECEIVER_TAG);
-		try {
-			ServerCalls.getInstance().downloadMedia(url, fileString, receiver);
-		} catch (IOException ioException) {
-			Log.e(TAG, Log.getStackTraceString(ioException));
-			result = false;
-			receiver.send(Constants.UPDATE_DOWNLOAD_ERROR, null);
-		} catch (WebconnectionException webconnectionException) {
-			Log.e(TAG, Log.getStackTraceString(webconnectionException));
-			result = false;
-			errorMessage = R.string.error_internet_connection;
-			receiver.send(Constants.UPDATE_DOWNLOAD_ERROR, null);
-		}
-
-		if (!result) {
-			showToast(errorMessage, true);
-			return;
-		}
-
-		showToast(R.string.notification_download_finished, false);
-	}
-
-	private void showToast(final int messageId, boolean error) {
-		if (error) {
-			handler.post(new Runnable() {
-
-				@Override
-				public void run() {
-					ToastUtil.showError(getBaseContext(), messageId);
-				}
-			});
-		} else {
-			handler.post(new Runnable() {
-
-				@Override
-				public void run() {
-					ToastUtil.showSuccess(getBaseContext(), messageId);
-				}
-			});
-		}
+		String programName = intent.getStringExtra(PROGRAM_NAME);
+		final ResultReceiver receiver = intent.getParcelableExtra(RECEIVER);
+		StatusBarNotificationManager manager = StatusBarNotificationManager.getInstance();
+		final int notificationId = manager.createDownloadNotification(getApplicationContext(), programName);
+		startForeground(notificationId,
```

Comment:
Regarding user experience (which we briefly talked about some time in the past, I believe): We do already have a download dialog. Showing an additional notification does not really provide a better experience. 

So basically, the question is, if we need a foreground service at all here (I know, [https://jira.catrob.at/browse/CATROID-168](CATROID-168) is about turning this service into a foreground service, but since a dialog is shown to the user anyway, we might just keep this service as a background service. 

So basically this is a question for a PO (@wslany) about how we want the media download to behave. 

Related to this question: How does the ui behave in case a user navigates back from the WebViewActivity (is this even possible?) and the media download has not yet finished? This might only be testable on an emulator when setting the connection to a very low speed.

Reply:
Another note: I'd be ok with this not being a foreground service, since the implications would not be problematic here.

Reply:
First of all I agree with you about changing/(closing and creating ) a  ticket for backgorund service for the MediaDownloadService as we talked about it weeks ago. But why now??? We talked weeks before, we could have asked @wslany before opening the PR. So the work/time I invested would be less than now.... 
We talked in the retroperspective about the lack of  communication in the team -> so pls next time we should talk thorugh it before anyone in the team does an PR and afterwards the conclusion  is "hmm yeah   pls implement it different and not as it is written in the ticket".

Reply:
First off, sorry for the late response, I was on vacation until today. 

Of course you are right, tickets should not be changed drastically when already being in CR. This ticket was not very well defined, so this situation is on me. 

However, changing the implementation from a foreground service back to a background service is only removing one LOC (and about five LOC that are unused after that), so this should not be much work. So basically the decision if we want a foreground service (with the implied progress notification additional to the progress dialog) or a background service (ui stays the same, but service may be stopped in case of app is moved to background) can be made during PO review. 

Note that I really do like the rest of the implementation and hope it can be merged soon, but there has to be a decision (which not I, but a PO should make) on this issue first.